soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -67,50 +82,88 @@ static ALWAYS_INLINE void clock_init(void)
 #if defined(CONFIG_SOC_LPC55S06) || defined(CONFIG_SOC_LPC55S16) || \
 	defined(CONFIG_SOC_LPC55S28) || defined(CONFIG_SOC_LPC55S36) || \
 	defined(CONFIG_SOC_LPC55S69_CPU0)
-    /*!< Set up the clock sources */
-    /*!< Configure FRO192M */
-	/*!< Ensure FRO is on  */
+	/* Set up the clock sources */
+	/* Configure FRO192M */
+	/* Ensure FRO is on  */
 	POWER_DisablePD(kPDRUNCFG_PD_FRO192M);
-	/*!< Set up FRO to the 12 MHz, just for sure */
+	/* Set up FRO to the 12 MHz, to ensure we can change the clock freq */
 	CLOCK_SetupFROClocking(12000000U);
-	/*!< Switch to FRO 12MHz first to ensure we can change the clock */
+	/* Switch to FRO 12MHz first to ensure we can change the clock */
 	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
 
-	/* Enable FRO HF(96MHz) output */
-	CLOCK_SetupFROClocking(96000000U);
-
-#ifdef CONFIG_INIT_PLL0
-	/*!< Ensure XTAL16M is on  */
-	PMC->PDRUNCFGCLR0 |= PMC_PDRUNCFG0_PDEN_XTAL32M_MASK;
-	PMC->PDRUNCFGCLR0 |= PMC_PDRUNCFG0_PDEN_LDOXO32M_MASK;
+	/* Configure XTAL32M */
+	CLOCK_SetupExtClocking(16000000U);
 
-	/*!< Ensure CLK_IN is on  */
+	/* Ensure CLK_IN is on  */
 	SYSCON->CLOCK_CTRL |= SYSCON_CLOCK_CTRL_CLKIN_ENA_MASK;
 	ANACTRL->XO32M_CTRL |= ANACTRL_XO32M_CTRL_ENABLE_SYSTEM_CLK_OUT_MASK;
 
-	/*!< Switch PLL0 clock source selector to XTAL16M */
+
+#if defined(CONFIG_SOC_LPC55S06) || !defined(CONFIG_INIT_PLL1)
+	SystemCoreClock = 96000000U;
+#else
```

Comment:
magic numbers

Reply:
I have added a comment that should help with all magic number comments.